src/core/hle/service/pctl/module.cpp

```diff
@@ -119,92 +133,273 @@ class IParentalControlService final : public ServiceFramework<IParentalControlSe
     }
 
 private:
+    bool CheckFreeCommunicationPermissionImpl() {
+        if (states.temporary_unlocked) {
+            return true;
+        }
+        if ((states.application_info.parental_control_flag & 1) == 0) {
+            return true;
+        }
+        if (pin_code[0] == '\0') {
+            return true;
+        }
+        if (!settings.is_free_communication_default_on) {
+            return true;
+        }
+        // TODO(ogniK): Check for blacklisted/exempted applications
+        return true;
+    }
+
+    bool ConfirmStereoVisionPermissionImpl() {
+        if (states.temporary_unlocked) {
+            return true;
+        }
+        if (pin_code[0] == '\0') {
+            return true;
+        }
+        if (!settings.is_stero_vision_restricted) {
+            return false;
+        }
+        return true;
+    }
+
+    void SetStereoVisionRestrictionImpl(bool is_restricted) {
+        if (settings.disabled) {
+            return;
+        }
+
+        if (pin_code[0] == '\0') {
+            return;
+        }
+        settings.is_stero_vision_restricted = is_restricted;
+    }
+
     void Initialize(Kernel::HLERequestContext& ctx) {
-        LOG_WARNING(Service_PCTL, "(STUBBED) called");
+        LOG_DEBUG(Service_PCTL, "called");
+        IPC::ResponseBuilder rb{ctx, 2};
+
+        if (False(capability & (Capability::Application | Capability::System))) {
+            LOG_ERROR(Service_PCTL, "Invalid capability! capability={:X}",
+                      static_cast<s32>(capability));
```

Comment:
```suggestion
                      capability);
```

fmt handles passed in enum class values, so casts are no longer necessary.